pkg/pool-weighted/contracts/lbp/LBPool.sol

```diff
@@ -0,0 +1,299 @@
+    // SafeCast.
+    struct PoolState {
+        uint32 startTime;
+        uint32 endTime;
+        uint64 startWeight0;
+        uint64 endWeight0;
+        bool swapEnabled;
+    }
+
+    // LBPs are constrained to two tokens.
+    uint256 private constant _NUM_TOKENS = 2;
+
+    // LBPools are deployed with the Balancer standard router address, which we know reliably reports the true
+    // originating account on operations. This is important for liquidity operations, as these are permissioned
+    // operations that can only be performed by the owner of the pool. Without this check, a malicious router
+    // could spoof the address of the owner, allowing anyone to call permissioned functions.
+
+    // solhint-disable-next-line var-name-mixedcase
+    address private immutable _TRUSTED_ROUTER;
+
+    PoolState private _poolState;
+
+    /**
+     * @notice Emitted when the owner enables or disables swaps.
+     * @param swapEnabled True if we are enabling swaps
+     */
+    event SwapEnabledSet(bool swapEnabled);
+
+    /**
+     * @notice Emitted when the owner initiates a gradual weight change (e.g., at the start of the sale).
+     * @dev Also emitted on deployment, recording the initial state.
+     * @param startTime The starting timestamp of the update
+     * @param endTime  The ending timestamp of the update
+     * @param startWeights The weights at the start of the update
+     * @param endWeights The final weights after the update is completed
+     */
+    event GradualWeightUpdateScheduled(
+        uint256 startTime,
+        uint256 endTime,
+        uint256[] startWeights,
+        uint256[] endWeights
+    );
+
+    /// @dev Indicates that the router that called the Vault is not trusted, so any operations should revert.
+    error RouterNotTrusted();
+
+    /// @dev Indicates that the `owner` has disabled swaps.
+    error SwapsDisabled();
+
+    constructor(
+        NewPoolParams memory params,
+        IVault vault,
+        address owner,
+        bool swapEnabledOnStart,
+        address trustedRouter
+    ) WeightedPool(params, vault) Ownable(owner) {
+        // WeightedPool validates `numTokens == normalizedWeights.length`, and ensures valid weights.
+        // Here we additionally enforce that LBPs must be two-token pools.
+        InputHelpers.ensureInputLengthMatch(_NUM_TOKENS, params.numTokens);
+
+        // Set the trusted router (passed down from the factory).
+        _TRUSTED_ROUTER = trustedRouter;
+
+        // solhint-disable-next-line not-rely-on-time
+        uint32 currentTime = block.timestamp.toUint32();
+        _startGradualWeightChange(currentTime, currentTime, params.normalizedWeights, params.normalizedWeights);
+        _setSwapEnabled(swapEnabledOnStart);
+    }
+
+    /// @notice Returns trusted router, which is the gateway to add liquidity to the pool.
+    function getTrustedRouter() external view returns (address) {
```

Comment:
We have this in two places (pool and factory). We could also just store the factory, and call `getTrustedRouter` on the factory. Though then we'd need a getter for the factory anyway, so I guess we're not saving anything (and would need an interface to call anything on the factory).

So I'd say it's fine as is unless we have another reason to keep the factory address. Maybe for the "factory as router" initialization?

Also there aren't tests for the getter in either the pool or the factory.